tchannel/tornado/peer.py

```diff
@@ -702,8 +702,18 @@ def choose(self, hostport=None, blacklist=None):
         """
 
         blacklist = blacklist or set()
+        # We we already have the connection in our peer list, use it. Else
+        # don't add it to our peer list.
+        # Refs https://github.com/uber/tchannel-python/issues/464
         if hostport:
-            return self.get(hostport)
+            if hostport in self.hosts:
```

Comment:
we might want to just use
```
if hostport in self._peers:
```
That way (I assume) we're not converting the _peers dict into a list of keys on every choose.